plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java

```diff
@@ -1401,7 +1405,7 @@ public void dropColumn(ConnectorSession session, ConnectorTableHandle tableHandl
 
         long commitVersion = table.getReadVersion() + 1;
         List<String> partitionColumns = metadataEntry.getOriginalPartitionColumns();
-        if (partitionColumns.contains(dropColumnName)) {
+        if (deltaLakeColumn.getColumnType() == PARTITION_KEY) {
```

Comment:
`TestDeltaLakeColumnMappingMode.testDropLastNonPartitionColumnWithColumnMappingMode` starts failing without this change.